librz/main/rz-diff.c

```diff
@@ -1339,6 +1428,749 @@ static bool rz_diff_graphs_files(DiffContext *ctx) {
+		rz_cons_canvas_write(canvas, line);
+
+		snprintf(line, lsize, "%sZ%s  increase the offset of the file0 by 1\n", number, reset);
+		rz_cons_canvas_gotoxy(canvas, 6, 8);
+		rz_cons_canvas_write(canvas, line);
+
+		snprintf(line, lsize, "%sA%s  decrease the offset of the file0 by 1\n", number, reset);
+		rz_cons_canvas_gotoxy(canvas, 6, 9);
+		rz_cons_canvas_write(canvas, line);
+
+		snprintf(line, lsize, "%sC%s  increase the offset of the file1 by 1\n", number, reset);
+		rz_cons_canvas_gotoxy(canvas, 6, 11);
+		rz_cons_canvas_write(canvas, line);
+
+		snprintf(line, lsize, "%sD%s  decrease the offset of the file1 by 1\n", number, reset);
+		rz_cons_canvas_gotoxy(canvas, 6, 12);
+		rz_cons_canvas_write(canvas, line);
+
+		snprintf(line, lsize, "%sM%s  next difference\n", number, reset);
+		rz_cons_canvas_gotoxy(canvas, 6, 14);
+		rz_cons_canvas_write(canvas, line);
+
+		snprintf(line, lsize, "%sN%s  previous difference\n", number, reset);
+		rz_cons_canvas_gotoxy(canvas, 6, 15);
+		rz_cons_canvas_write(canvas, line);
+
+		snprintf(line, lsize, "%sG%s  sets both offsets to a common value\n", number, reset);
+		rz_cons_canvas_gotoxy(canvas, 6, 16);
+		rz_cons_canvas_write(canvas, line);
+
+		snprintf(line, lsize, "%s%s%s decrease both offsets by %u\n", number, arrow_up, reset, 1 << shift);
+		rz_cons_canvas_gotoxy(canvas, 6, 18);
+		rz_cons_canvas_write(canvas, line);
+
+		snprintf(line, lsize, "%s%s%s increase both offsets by %u\n", number, arrow_down, reset, 1 << shift);
+		rz_cons_canvas_gotoxy(canvas, 6, 19);
+		rz_cons_canvas_write(canvas, line);
+
+		snprintf(line, lsize, "%s%s%s increase both offsets by 1\n", number, arrow_left, reset);
+		rz_cons_canvas_gotoxy(canvas, 6, 20);
+		rz_cons_canvas_write(canvas, line);
+
+		snprintf(line, lsize, "%s%s%s decrease both offsets by 1\n", number, arrow_right, reset);
+		rz_cons_canvas_gotoxy(canvas, 6, 21);
+		rz_cons_canvas_write(canvas, line);
+
+		snprintf(line, lsize, "%s:%s  seek at offset (relative via +-)\n", number, reset);
+		rz_cons_canvas_gotoxy(canvas, 6, 22);
+		rz_cons_canvas_write(canvas, line);
+	}
+
+	rz_cons_canvas_print(canvas);
+	rz_cons_gotoxy(width, height);
+	rz_cons_flush();
+	return true;
+}
+
+static char *visual_prompt(DiffHexView *hview, const char *prompt) {
+	char buf[1024];
+	rz_cons_gotoxy(0, hview->screen.height);
+	rz_cons_clear_line(0);
+	rz_cons_printf("%s%s ", hview->colors.reset, prompt);
+	rz_line_set_prompt(":> ");
+	rz_cons_flush();
+	rz_cons_fgets(buf, sizeof(buf), 0, NULL);
+	if (*buf) {
+		return strdup(buf);
+	}
+	return NULL;
+}
+
+static void prompt_offset_and_seek(DiffHexView *hview, ut64 minseek) {
+	char *value = visual_prompt(hview, " you can input an absolute offset or a relative offset by adding the prefix + or -\n offset");
+	if (value) {
+		const char *p = rz_str_trim_head_ro(value);
+		if (!IS_DIGIT(*p) && *p != '-' && *p != '+') {
+			free(value);
+			return;
+		}
+		st64 number = strtoll((*p == '+') ? p + 1 : p, NULL, 0);
```

Comment:
Maybe better to use `rz_num` here?

Reply:
no, because it's too messy and also calls the same function